test/distributed/tensor/parallel/test_2d_parallel.py

```diff
@@ -127,6 +167,34 @@ def test_2d_fsdp_integration_functionality(self) -> None:
         )
         self.assertFalse(is_nested_tensor(optim_state["state"]["net3.bias"]["exp_avg"]))
 
+    @with_comms
+    @skip_if_lt_x_gpu(4)
+    def test_2d_with_ddp(self) -> None:
+        model_tp = init_model(use_ddp=True)[0]
+        optim = torch.optim.Adam(model_tp.parameters(), lr=0.0001)
+
+        # Create Input
+        input_seed = self.rank
+        torch.manual_seed(input_seed + 1)
+        input = torch.rand(4, 5).cuda(self.rank)
+
+        model_tp(input).sum().backward()
+        optim.step()
+        sd = model_tp.state_dict()
+        load_tp = init_model(use_ddp=True)[0]
+        from torch.distributed._tensor import DTensor
+        for p in load_tp.parameters():
+            if isinstance(p, DTensor):
+                p._local_tensor.data.zero_()
+            else:
+                p.data.zero_()
+
+        load_tp.load_state_dict(sd)
```

Comment:
I appreciate that we keep save/load in mind, but what's more important is the correct of gradient and we are updating model correctly.